### PR TITLE
A few assorted cleanups from the statistics branch

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

These are a few assorted cleanups of the statistics branch, which are needed for but basically independent of the statistics. The point of this PR is to simplify code-review and decrease the size of the statistics PR.

### Related issues:

Preparation commit for #3052.

### Release note:

Not yet.

### Documentation change:

Not yet.

### CODE REVIEW

**Comment by dirkhh** [comment]
The Tumbleweed error was an openSUSE package download error. Restarting.
There are a hand full more of prep and cleanup and build-system fix and GitHub Action commits that could also quite reasonably pulled out - even if they are related to QtCharts (well, quite a few of them are).
Thoughts? That would make it easier to simply shrink the sheer number of commits in #3052

File: core/divelist.c [hunk 1]
```diff
@@ -823,7 +819,7 @@ void process_loaded_dives()
 
 	fulltext_populate();
 
-	/* Inform frontend of reset data. This should reset all the models. */
+	/* Inform frontend of reset data. This should reset all the models and the filter. */
 	emit_reset_signal();
```
Comment (dirkhh) [inline core/divelist.c]: is there something implicit that I'm missing - I don't see a `connect` call of that new `reset()` slot to the signal...

**Review by dirkhh — APPROVED** [review]
One thing I'm confused about, other than that this looks good.

File: core/divelist.c (hunk 1 above)
Comment (bstoeger) [reply core/divelist.c]: Good catch - this comment change is an artifact of a previous version. Removed.

**Comment by bstoeger** [comment]
> There are a hand full more of prep and cleanup and build-system fix and GitHub Action commits that could also quite reasonably pulled out - even if they are related to QtCharts (well, quite a few of them are).
Thoughts? That would make it easier to simply shrink the sheer number of commits in #3052

Sure, I can break these out too.

**Review by dirkhh — APPROVED** [review]

**Pull request merged**